src/operator/src/statement/ddl.rs

```diff
@@ -1726,48 +1726,42 @@ fn find_partition_entries(
     partitions: &Option<Partitions>,
     partition_columns: &[String],
     query_ctx: &QueryContextRef,
-) -> Result<Vec<Vec<PartitionBound>>> {
-    let entries = if let Some(partitions) = partitions {
-        // extract concrete data type of partition columns
-        let column_defs = partition_columns
-            .iter()
-            .map(|pc| {
-                create_table
-                    .column_defs
-                    .iter()
-                    .find(|c| &c.name == pc)
-                    // unwrap is safe here because we have checked that partition columns are defined
-                    .unwrap()
-            })
-            .collect::<Vec<_>>();
-        let mut column_name_and_type = HashMap::with_capacity(column_defs.len());
-        for column in column_defs {
-            let column_name = &column.name;
-            let data_type = ConcreteDataType::from(
-                ColumnDataTypeWrapper::try_new(column.data_type, column.datatype_extension)
-                    .context(ColumnDataTypeSnafu)?,
-            );
-            column_name_and_type.insert(column_name, data_type);
-        }
+) -> Result<Vec<PartitionExpr>> {
+    let Some(partitions) = partitions else {
+        return Ok(vec![]);
+    };
 
-        // Transform parser expr to partition expr
-        let mut partition_exprs = Vec::with_capacity(partitions.exprs.len());
-        for partition in &partitions.exprs {
-            let partition_expr =
-                convert_one_expr(partition, &column_name_and_type, &query_ctx.timezone())?;
-            partition_exprs.push(vec![PartitionBound::Expr(partition_expr)]);
-        }
+    // extract concrete data type of partition columns
+    let column_defs = partition_columns
+        .iter()
+        .map(|pc| {
+            create_table
+                .column_defs
+                .iter()
+                .find(|c| &c.name == pc)
+                // unwrap is safe here because we have checked that partition columns are defined
+                .unwrap()
+        })
+        .collect::<Vec<_>>();
+    let mut column_name_and_type = HashMap::with_capacity(column_defs.len());
+    for column in column_defs {
+        let column_name = &column.name;
+        let data_type = ConcreteDataType::from(
+            ColumnDataTypeWrapper::try_new(column.data_type, column.datatype_extension)
+                .context(ColumnDataTypeSnafu)?,
+        );
+        column_name_and_type.insert(column_name, data_type);
+    }
```

Comment:
maybe save a temporary vec:

```suggestion
    let column_name_and_type = partition_columns
        .iter()
        .map(|pc| {
            let column = create_table
                .column_defs
                .iter()
                .find(|c| &c.name == pc)
                // unwrap is safe here because we have checked that partition columns are defined
                .unwrap();
            let column_name = &column.name;
            let data_type = ConcreteDataType::from(
                ColumnDataTypeWrapper::try_new(column.data_type, column.datatype_extension)
                    .context(ColumnDataTypeSnafu)?,
            );
            (column_name, data_type)
        })
        .collect::<HashMap<_, _>>();
```